Library/Homebrew/utils/gems.rb

```diff
@@ -151,6 +154,33 @@ def install_bundler!
     ENV["BUNDLER_VERSION"] = old_bundler_version
   end
 
+  def user_gem_groups
+    @user_gem_groups ||= if GEMGROUPS_FILE.exist?
+      GEMGROUPS_FILE.readlines(chomp: true)
+    else
+      # Backwards compatibility. This else block can be replaced by `[]` by the end of 2023.
+      require "settings"
+      groups = Homebrew::Settings.read(:gemgroups)&.split(";") || []
+      write_user_gem_groups(groups)
+      Homebrew::Settings.delete(:gemgroups)
+      groups
+    end
+  end
+
+  def write_user_gem_groups(groups)
+    GEMGROUPS_FILE.write(groups.join("\n"))
+  end
+
+  def forget_user_gem_groups!
+    if GEMGROUPS_FILE.exist?
+      GEMGROUPS_FILE.truncate(0)
+    else
+      # Backwards compatibility. This else block can be removed by the end of 2023.
+      require "settings"
+      Homebrew::Settings.delete(:gemgroups)
```

Comment:
Should this delete from here too even if the `GEM_GROUPS_FILE` exists if the setting also exists?

Reply:
The reading code above should handle the initial creation of the `GEM_GROUPS_FILE` and delete the old setting, so there shouldn't be two existing.

When we remove this code, there will be a dangling setting left for any latecomers but that's relatively minor and there's many other settings we haven't cared about.